core/arch/instrlist.h

```diff
@@ -116,6 +121,22 @@ instrlist_get_translation_target(instrlist_t *ilist);
 void
 instrlist_set_our_mangling(instrlist_t *ilist, bool ours);
 
+DR_API
+/**
+ * All future instructions inserted into \p ilist will be predicated
+ * with \p pred. This is a convenience routine to make it easy to have
+ * emitted code from internal dr components predicated.
+ *
+ * \note only has an effect on ARM
+ */
+void
+instrlist_set_auto_predicate(instrlist_t *ilist, dr_pred_type_t pred);
```

Comment:
WDYT about having drmgr auto-set this unless the client requests not to?  Or do you think no more automation is needed?  I think that for simple clients they would appreciate this being done for them auto-magically.  More sophisticated ones like the memory tracers can disable the predicate for their instr tracing and leave the auto default for the data tracing.  (Seems fine to separate this automation as a separate CL from what you have here, just wanted to know the direction you were thinking.)